src/frontend/staff-portal/src/app/components/staff-workbench/update-request-inbox/update-request-inbox.component.ts

```diff
@@ -52,55 +52,40 @@ export class UpdateRequestInboxComponent implements OnInit, AfterViewInit {
     this.getAllDisputesWithPendingUpdates();
   }
 
-  isNew(d: Dispute): boolean {
-    return d.status == DisputeStatus.New && (d.emailAddressVerified === true || !d.emailAddress);
-  }
-
   getAllDisputesWithPendingUpdates(): void {
     this.logger.log('UpdateRequestInboxComponent::getAllDisputesWithPendingUpdates');
 
-    this.disputes = [];
+    this.disputesWithUpdates = [];
 
-    this.dataSource.data = this.disputes;
+    this.dataSource.data = this.disputesWithUpdates;
 
     this.disputeService.getDisputesWithPendingUpdates().subscribe((response) => {
       this.logger.info(
         'UpdateRequestInboxComponent::getAllDisputesWithPendingUpdates response',
         response
       );
 
-      response.forEach((d:Dispute) => {
-        if (d.status != "CANCELLED") { // do not show cancelled
-          var newDispute: Dispute = {
-            ticketNumber: d.ticketNumber,
-            disputantSurname: d.disputantSurname,
-            disputantGivenNames: d.disputantGivenNames,
-            disputeId: d.disputeId,
-            userAssignedTo: d.userAssignedTo,
-            emailAddressVerified: d.emailAddressVerified,
-            emailAddress: d.emailAddress,
-            __CourtHearing: false,
-            __DateSubmitted: new Date(d.submittedTs),
-            __FilingDate: d.filingDate != null ? new Date(d.filingDate) : null,
-            __UserAssignedTs: d.userAssignedTs != null ? new Date(d.userAssignedTs) : null,
-            additionalProperties: d.additionalProperties,
-            status: d.status,
-            __RedGreenAlert: d.status == DisputeStatus.New ? 'Green' : '',
-            userAssignedTs: d.userAssignedTs
-          }
-
-          // set court hearing to true if its true for any one of the three possible counts
-          // otherwise false
-          if (d.disputeCounts) d.disputeCounts.forEach(c => {
-            if (c.requestCourtAppearance === this.RequestCourtAppearance.Y) {
-              newDispute.__CourtHearing = true;
-            }
-          });
-
-          this.disputes = this.disputes.concat(newDispute);
+      response.forEach((d:DisputeWithUpdates) => {
+        var newDispute: DisputeWithUpdates = {
```

Comment:
Is there any reason to create a new object instead of modifying the original one?
And given names can join inside the service and reuse the function

Reply:
Input, output can be changed to 
Dispute | DisputeForUpdates

Reply:
it is functional the way i have it